src/call/mod.rs

```diff
@@ -108,6 +110,142 @@ impl RpcStatus {
     }
 }
 
+/// `MessageReader` is a zero-copy reader for the message payload.
+///
+/// To achieve zero-copy, use the BufRead API `fill_buf` and `consume`
+/// to operate the reader.
+pub struct MessageReader {
+    buf: *mut GrpcByteBuffer,
+    reader: GrpcByteBufferReader,
+    slice: Option<GrpcSlice>,
+    // Hack: apparently its lifetime depends on `self.slice`. However
+    // it's not going to be accessed by others directly, hence it's safe
+    // to mark it static here.
+    bytes: &'static [u8],
+    length: usize,
+    empty: bool,
```

Comment:
This is confusing. As a reader, it should only have two states, like EOF is not reached and EOF is reached. As stated above, `Option<MessageReader>` is not the same as `MessageReader`, `None` is not empty, it means the reader doesn't exist at all. Let's make it clear.

Reply:
`empty` here takes the place of `Option`. Now I'm somewhat changed my mind -- `Option` is now good to me.